### PR TITLE
Fixed title property (article-title) and add page title property

### PR DESCRIPTION
**Pull request by wachterjohannes**
| Q | A
| --- | ---
| Bug fix? | no
| New feature? | no
| BC breaks? | no
| Deprecations? | no
| Fixed tickets | none
| Related issues/PRs | build ontop #122 requires https://github.com/sulu/sulu-document-manager/pull/110 and https://github.com/sulu/sulu/pull/3306
| License | MIT
| Documentation PR | none

#### What's in this PR?

This PR fixes the property title (which should contain the article-title) and add a new property page-title.

#### To Do

- [x] Tests

### CODE REVIEW

File: Document/Subscriber/ArticlePageSubscriber.php [hunk 1]
```diff
@@ -123,6 +130,70 @@ public function setWorkflowStageOnArticle(PersistEvent $event)
...
+        if (!$document instanceof ArticlePageDocument || $event->hasNode()) {
+            return;
+        }
+
+        $pageTitle = $this->getPageTitle($document);
+        $nodeName = $this->slugifier->slugify($pageTitle ?: uniqid('page-', true));
+        $nodeName = $this->resolver->resolveName($event->getParentNode(), $nodeName);
+
+        // jackrabbit can not handle node-names which contains a number followed by "e" e.g. 10e
+        $nodeName = preg_replace('((\d+)([eE]))', '$1-$2', $nodeName);
```
Comment (alexander-schranz) [inline Document/Subscriber/ArticlePageSubscriber.php]: shouldnt we make a component in the sulu document manager repository for that?

File: Document/Subscriber/ArticlePageSubscriber.php [hunk 2]
```diff
@@ -123,6 +130,70 @@ public function setWorkflowStageOnArticle(PersistEvent $event)
...
+     */
+    public function setNodeOnPersist(PersistEvent $event)
+    {
+        $document = $event->getDocument();
+        if (!$document instanceof ArticlePageDocument || $event->hasNode()) {
+            return;
+        }
+
+        $pageTitle = $this->getPageTitle($document);
+        $nodeName = $this->slugifier->slugify($pageTitle ?: uniqid('page-', true));
```
Comment (alexander-schranz) [inline Document/Subscriber/ArticlePageSubscriber.php]: add comment why

File: Document/Subscriber/ArticlePageSubscriber.php (hunk 1 above)
Comment (wachterjohannes) [inline Document/Subscriber/ArticlePageSubscriber.php]: this is currently only included in `AutonameBehavior` i have currently no idea where to put this code instead.

**Comment by wachterjohannes** [comment]
@alexander-schranz fixed

File: Document/Subscriber/ArticlePageSubscriber.php [hunk 3]
```diff
@@ -123,6 +130,72 @@ public function setWorkflowStageOnArticle(PersistEvent $event)
...
+        $document = $event->getDocument();
+        if (!$document instanceof ArticlePageDocument || $event->hasNode()) {
+            return;
+        }
+
+        $pageTitle = $this->getPageTitle($document);
+
+        // if no page-title exists use a unique-id
+        $nodeName = $this->slugifier->slugify($pageTitle ?: uniqid('page-', true));
+        $nodeName = $this->resolver->resolveName($event->getParentNode(), $nodeName);
```
Comment (alexander-schranz) [inline Document/Subscriber/ArticlePageSubscriber.php]: should this not be called after the `preg_replace`?

File: Document/Subscriber/ArticlePageSubscriber.php (hunk 1 above)
Comment (alexander-schranz) [inline Document/Subscriber/ArticlePageSubscriber.php]: @wachterjohannes maybe we can create a NodeNameSlugifier or something and call there also the slugify? I know its just one line but think this logic should not be copied.

**Comment by wachterjohannes** [comment]
@alexander-schranz as soon as this https://github.com/sulu/sulu-document-manager/pull/110 is merged you can review this PR again

**Pull request merged**